gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/QueryServiceEntityFetcher.java

```diff
@@ -421,14 +421,16 @@ public EntityFetcherResponse getTimeAggregatedMetrics(
               QueryAndGatewayDtoConverter.convertQueryValueToGatewayValue(
                   row.getColumn(idColumns.size()));
           if (value.getValueType() == ValueType.STRING) {
-            long time = Long.parseLong(value.getString());
-            intervalBuilder.setStartTimeMillis(time);
-            intervalBuilder.setEndTimeMillis(time + TimeUnit.SECONDS.toMillis(periodSecs));
+            long startTime = Long.parseLong(value.getString());
```

Comment:
A bit of an aside: The fact that avg rate was implemented in gateway has always been tech debt - that responsibility to evaluate a function belongs to query service (which becomes even more relevant as we consider adding a metric store). Any thought on level of effort to moving it? It may make sense to do as part of a broader effort to remove _any_ pinot knowledge from gateway - of which there's quite a bit today.

Reply:
This makes sense. I guess it shouldn't be a big effort for avg_rate. We have to clean up the translation mapping from avg_rate to SUM and move the calculation to query service. I have created the issue for this specific case - https://github.com/hypertrace/hypertrace/issues/296, and we will take it up as follow up. I need to scan for other pinot deps and will move to its own ticket.